[11.0][MIG] account_payment_order

etobella opened the pull request:
Migration to 11.0
Still working on tests and multicompany environments

Depends on

- [x] account_payment_mode #399 
- [x] account_payment_partner #400

alexis-via committed:
Start to port bank-payment to v9 (with a lot of improvements) during the Sorrento Code sprint 2016

Improvements include:
- full re-organisation of modules and big re-organisation of the code
- simplification of the code related to the fact that support for direct debit is now in t
he base module, not added by an optional module account_direct_debit (module was removed)
- new design of the wizard to select move lines to pay
- support for non-SEPA file transfer-
- support for German direct debit SEPA files (fixes bug #129)
- remove workflow of payment.order

This port to v9 is not finished... there is still a lot of work:
- finish the code of account_payment_order/wizard/account_payment_line_create.py
- port account_banking_payment_transfer and integrate it inside account_payment_order
- fix bugs
- clean-up code, remove dead code
- test in several complex scenarios

alexis-via committed:
Finalise the wizard of selection of move lines to pay

Add button "Add to payment/debit order" on invoice form view
Started to integrate payment transfer in account_payment_order (not finished at all though)
Various fixes/changes/improvements/...

alexis-via committed:
Finalise merge of account_banking_payment_transfer into account_payment_order

Add support for transfer moves
Display transfer moves in a dedicated tab in payment order view
Code cleanup

alexis-via committed:
Take into account the review of @yvaucher

alexis-via committed:
PEP8 compliant with the latest version of flake8

alexis-via committed:
Filter on allowed_journal_ids even if onchange of payment_mode_id is not played (for example when the payment line is pushed from the invoice)

alexis-via committed:
Add option 'target_move' (all or posted) in wizard to select move lines to pay

Remove module account_payment_include_draft_move

alexis-via committed:
Copy payment mode and mandate from invoice to refund

Run dos2unix on the few files that had DOS carriage return

alexis-via committed:
Add module account_banking_mandate_sale

Add option 'mandate_required' on payment orders
Autoselect first valid mandate on customer invoice that have a payment mode 'mandate_required' = True
Add option on select move lines to pay wizard to allow selection of litigation moves (unchecked by default), in order to integrate the feature of the module account_payment_blocking

alexis-via committed:
Remove 'done' state on payment orders

alexis-via committed:
Payment date should never be in the past.

alexis-via committed:
Move fields mandate_required and export_ascii from payment mode to payment method

Display chatter on payment orders
Several small usability improvements

alexis-via committed:
Avoid empty group in payment mode

Code cleanup in views

alexis-via committed:
Add support for using bank account as offsetting account in journal entry

alexis-via committed:
Add option no_debit_before_maturity on payment mode

alexis-via committed:
Take into accounts the remarks of Eric Caudal

alexis-via committed:
Add hooks for l10n_ch_sepa (pain.001.001.03.ch.02)

Move code of account_banking_pain_base/models/banking_export_pain.py to account_banking_pain_base/models/account_payment_order.py
Allow cancel when payment order is in "File Generated" state
Add postal address in party block

alexis-via committed:
Don't group payment lines that use structured communication

alexis-via committed:
Add wizard to select several invoices at once and add them to payment order

Fix bug in groupby on bank payment lines

sbidoul committed:
Avoid confusion between payment mode and payment method

alexis-via committed:
Move field bank_account_required from module account_payment_partner to account_payment_mode

Make the mandate a required field on payment line when the payment method has mandate_required=True
Make the bank account a required field on payment line when the payment method has bank_account_required=True
Minor code cleanup
PEP8

alexis-via committed:
FIX manual payment method

alexis-via committed:
Enable the payment methods by default on bank journals (including existing bank journals via post_install scripts)

Enable "Generate Accounting Entries On File Upload" by default

sbidoul committed:
[IMP] prevent deleting a move-line that is linked to a payment order line

sbidoul committed:
[FIX] better filtering of move lines already included in open payment order

alexis-via committed:
Add support for multi-currency in the generation of the account move.

Take into account all the remaining remarks of Frederic Clementi dated May 30:
- see multi-currency amounts in the wizard to add transaction lines
- add help msg on bank payment lines

pedrobaeza committed:
[IMP] account_payment_order: Remove useless migration script

oca-transbot committed:
OCA Transbot updated translations from Transifex

yvaucher committed:
Move the account_payment_mode ir.rule in account_payment_mode module

sbidoul committed:
[IMP] account_payment_order: extensibility of _prepare_move

Pass the list of bank lines to _prepare_move so it is possible to customize the move (eg it's name) based on the lines being paid.

sbidoul committed:
[FIX] payment order: add missing onchange dependencies in move selection wizard

sbidoul committed:
[FIX] payment order: remove duplicate in previous commit

alexis-via committed:
FIX crash when selecting a customer that has a direct debit payment mode but no mandate

Hide many fields when payment_order_ok is False on a payment_mode
Fix payment line communication when the move line is not linked to an invoice (problem found when porting donation_debit_debit to v9)

alexis-via committed:
For the communication field, if self.move_id.ref is False, fallback on self.move_id.name

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

sbidoul committed:
[IMP] set partner on offsetting account move line if possible

fixes #277

sbidoul committed:
bump minor version of account_payment_order

oca-transbot committed:
OCA Transbot updated translations from Transifex

yvaucher committed:
Reactivate company rules in account_payment_order

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

Commit:
Port almost all modules to v10 (#305)

Port almost all modules to v10

* Update to EPC Rulebook v9.2 that start to apply on 2016-11-20 (bug #300)

Commit:
[FIX] account_payment_order: Use 'index' instead of 'select'

Commit:
[FIX] account_payment_order: Use 'index' instead of 'select'

jalzaga committed:
Remove internal_type=other restriction for transfer account (#311)

pedrobaeza committed:
Remove author as the contrib is only an small fix

alexis-via committed:
Creation/Modification of bank info only for Account Payment group

vrenaville committed:
[FIX] fill date instead off maturity date on account move line generated

mdietrichc2c committed:
Fix 2 issues on form view for payment order

pedrobaeza committed:
[CHG] *: Change copyright and author after company merging

alexis-via committed:
[FIX] Bug #353 crash when selecting a payment mode that has a variable link to a bank account

alexis-via committed:
Add 'post_move' option on account.payment.mode

alexis-via committed:
post_move default True, to keep the previous behavior

Bump up version number

alexis-via committed:
Add field default_date_prefered on payment mode

alexis-via committed:
Update version number

alexis-via committed:
[IMP] Explicit error when bank account is missing on bank journal

alexis-via committed:
Update version number

alexis-via committed:
'date_prefered' is a required field on payment orders: so, if 'date_prefered' key is present in vals but has no value, we use the default value on the payment mode

GillesTephaneMeyomesse committed:
[ADD] description field for payment order

mdietrichc2c committed:
[IMP] account_payment_order: Don't allow deletion of an uploaded payment order + allow to cancel an uploaded payment order

grindtildeath committed:
[IMP] account_payment_order: Restriction + tests

* Restrict the deletion of bank payment line
* Add test_cancel_payment_order

pedrobaeza committed:
[FIX] account_payment_order: Adapt test to v10

oca-transbot committed:
OCA Transbot updated translations from Transifex

GillesTephaneMeyomesse committed:
[MIG] add fr translation

Commit:
[IMP] account_payment_order: select partner bank account automatically.

grindtildeath committed:
[10.0]Fix payment method onchange with multicompany (#374)

GillesTephaneMeyomesse committed:
[ADD] report for payment order (#370)

etobella committed:
[MIG] account_payment_order

rgarnau commented on account_payment_order/README.rst:
```diff
@@ -0,0 +1,86 @@
+Configuration
+=============
+
+This module adds several options on Payment Modes, cf Accounting > Configuration > Management > Payment Modes.
```
'Invoicing' instead of 'Accounting'

rgarnau commented on account_payment_order/README.rst:
```diff
@@ -0,0 +1,86 @@
+Usage
+=====
+
+You can create a Payment Order via the menu Accounting > Payments > Payment Orders and then select the move lines to pay.
```
Invoicing

rgarnau commented on account_payment_order/README.rst:
```diff
@@ -0,0 +1,86 @@
+
+You can create a Payment Order via the menu Accounting > Payments > Payment Orders and then select the move lines to pay.
+
+You can create a Debit Order via the menu Accounting > Payments > Debit Orders and then select the move lines to debit.
```
Invoicing

rgarnau commented on account_payment_order/README.rst:
```diff
@@ -0,0 +1,86 @@
+
+This module also adds a button *Add to Payment Order* on supplier invoices and a button *Add to Debit Order* on customer invoices.
+
+You can print a Payment Order via the menu Accounting > Payments > Payment Orders and then select the payment oder to print.
```
Invoicing

rgarnau reviewed (commented):
LGTM :+1: (minor comments inline)

JordiBForgeFlow commented on account_payment_order/wizard/account_payment_line_create.py:
```diff
@@ -0,0 +1,155 @@
+        elif self.order_id.payment_type == 'inbound':
+            domain += [
+                ('debit', '>', 0),
+                ('account_id.internal_type', '=', 'receivable'),
```
My concern with payment.orders is that it does not seem to be able to handle refunds. So, in the scenario where you have an angry customer that expects a refund (and is not likely to come again - so little chance to reconcile the refund in the future with a new invoice), you cannot create a payment order, due to this added domain. 

IMHO what should only matter here is that with inbound payment orders you pay debit > 0, for receivables and payables, and with outbound payment orders you pay credit > 0, for receiv & payab.

cc @alexis-via @pedrobaeza @etobella

JordiBForgeFlow committed:
When the user looks for open payables or receivables, in the
context of payment orders, she should ocus on the amount that
is due to be paid. In this method we are forcing to display both
the amount due in company and in the invoice currency.

We then hide the fields debit and credit, because they add no value.

JordiBForgeFlow reviewed (approved):
:+1: Technical and functional review.

HviorForgeFlow commented on account_payment_order/README.rst:
```diff
@@ -0,0 +1,86 @@
+
+This module also adds a button *Add to Payment Order* on supplier invoices and a button *Add to Debit Order* on customer invoices.
+
+You can print a Payment Order via the menu Invoicing/Accounting > Payments > Payment Orders and then select the payment oder to print.
```
order

HviorForgeFlow commented on account_payment_order/models/account_invoice.py:
```diff
@@ -0,0 +1,111 @@
+                    "No Journal Entry on invoice %s") % inv.number)
+            if not inv.payment_order_ok:
+                raise UserError(_(
+                    "The invoice %s has a payment mode '%s' "
```
no string formatting:
"which is not selectable in payment orders.") % (inv.number, inv.payment_mode_id.display_name))

HviorForgeFlow reviewed (approved):
LGTM :+1:

angelmoya commented:
:+1:
